component/azstorage/azstorage.go

```diff
@@ -331,6 +333,10 @@ func (az *AzStorage) StreamDir(options internal.StreamDirOptions) ([]*internal.O
 	// increment streamdir call count
 	azStatsCollector.UpdateStats(stats_manager.Increment, streamDir, (int64)(1))
 
+	//check for filters provided
+	if len(az.stConfig.blobFilter) > 0 { //only apply filter if user has given
```

Comment:
After parsing we might have created a filter object so instead of checking the string length here rather we can check if that object is nil or not, will be a faster check.